onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1923,6 +1963,563 @@ Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-
+          output_shape.add_dim();
+        }
+        updateOutputShape(ctx, 0, output_shape, TypeProto::kSparseTensorType);
+      });
+
+  static const char* Binarizer_doc = R"DOC(
+This is similar to https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.Binarizer
+The presence of NaN will cause it to return an error.
+The operator will accept a single sparse tensor input and will output
+a sparse tensor of the same shape and type. For each of the elements it
+would compare the value of the 'threshold' attribute and for all values above the threshold
+it would set the output value for the same index to T(1) and T(0) otherwise.
+NaN 
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Binarizer)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "A sparse matrix of numeric type", "T")
+      .Attr(
+          "threshold",
+          "A boundary value",
+          AttributeProto::FLOAT,
+          static_cast<float>(0))
+      .Output(0, "Y", "A sparse tensor of the same type as input with 1 or 0 for each of the input values", "T")
+      .TypeConstraint("T",
+                      {"sparse_tensor(float)",
+                       "sparse_tensor(double)",
+                       "sparse_tensor(int64)",
+                       "sparse_tensor(int32)"},
+                      "The input must be a tensor of a numeric type. The output will be of the same tensor type.")
+      .SetDoc(Binarizer_doc)
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        propagateShapeFromInputToOutput(ctx, 0, 0);
+      });
+
+  static const char* OneHotEncoder_doc = R"DOC(
+The operator implements a simplified version of sklearn OneHotEncoder
+https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html
+`categories` attribute contains an array of category values.
+
+The input is a dense tensor of values.
+
+The operator produces a sparse output with a dense shape that has one more dimension than input.
+That dimension contains sparsified version of 1-of-K category output.
+
+Example:
+categories = [1021, 2002]
+dense input = [1021, 2002, 1023] dense shape: {3}
+
+Output dense shape = {3, 2} // 2 is the number of categories, choices for the last dimension include
+{0, 0}, {1, 0}, {0, 1}
+
+{0, 0} is a combination that would either match no categories or it is absent from the input.
+It will, therefore, be absent in the sparse output.
+
+For the rest of the values, we places ones in the following 2-D indices:
+(0, 0), (1, 1) and 1023 can not be categorized.
+
+The sparse output will have ones at the following flat COO indices: 0 and 4 (1 x 3 + 1)
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(OneHotEncoder)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "1-D dense tensor containing values to be categorized", "T")
+      .Output(0, "Y", "1-of-K 2-D sparse tensor output", "T1")
+      .Attr(
```

Comment:
Would scalar be supported as well? The dense version of OneHotEncoder (https://github.com/onnx/onnx/blob/master/docs/Operators-ml.md#ai.onnx.ml.OneHotEncoder) supports scalar tensor, and want to confirm this version would also support scalar as input.

Reply:
I have not implemented scalar support in SparseTensors and in this op. However, because the input is dense, we can  output shape {1] sparse tensor in the future.

Reply:
I can probably add scalar support to this one today.